p8e-api/src/main/kotlin/io/provenance/engine/grpc/v1/ObjectGrpc.kt

```diff
@@ -43,24 +44,31 @@ class ObjectGrpc(
         P8eMDC.set(publicKey(), clear = true)
 
         val msg = request.message.toByteArray()
-        val encryptionKeyPair = transaction { affiliateService.getEncryptionKeyPair(publicKey()) }
-        val signingKeyPair = transaction { affiliateService.getSigningKeyPair(publicKey()) }
+        val signer = transaction { affiliateService.getSigner(publicKey()) }
         val affiliateShares = transaction { affiliateService.getSharePublicKeys(request.toAudience().plus(publicKey())) }
+        val encryptionKeyRef = transaction { affiliateService.getEncryptionKeyRef(publicKey()) }
 
-        // Update the dime's audience list to use encryption public keys.
+        // Update the dime's audience list to use the encryption key ref, so we has accessibility
+        // to the audience's UUID as well, if SmartKey is used.
         val audience = request.toAudience().plus(affiliateShares.value).map {
             transaction {
                 try {
-                    affiliateService.getEncryptionKeyPair(it).public
+                    affiliateService.getEncryptionKeyRef(it)
```

Comment:
Can't this now be getEncryptionPublicKeys() since we changed the encryption side to not need the private keys at all?

Reply:
Yea, reverting back